### PR TITLE
Chapter 4 & 5 

### PR DESCRIPTION
**Pull request by EvonneNgXinYi**

### CODE REVIEW

**Comment by EvonneNgXinYi** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  line-height: 1.7em;
+}
+
+/* header */
+
+#logo {
+  float: left;
+  margin-right: 10px;
+  font-size: 1.7em;
+  color: #fff;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: you should write color code to another .scss file and import it to this file.

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    &:hover {
+    color: #222;
+    }
+  }
+  small {
+    float: left;
+  }
+  ul {
+    float: right;
+    list-style: none;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: each block should has one empty line among them

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/users.scss]: delete the files that never used

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Comment (saobv-2223) [inline app/helpers/users_helper.rb]: delete the files that never used

File: app/views/layouts/_footer.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <small>
+    The <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %></a>
+    by <%= link_to "Michael Hartl", Settings.links.rails_tutorial.author %></a>
```
Comment (saobv-2223) [inline app/views/layouts/_footer.html.erb]:
```suggestion
    The <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>
    by <%= link_to "Michael Hartl", Settings.links.rails_tutorial.author %>
```

File: app/views/layouts/_footer.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About", about_path %></li>
+      <li><%= link_to "Contact", '#' %></li>
```
Comment (saobv-2223) [inline app/views/layouts/_footer.html.erb]: use " please

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -1,7 +1,11 @@
+<div class="center jumbotron">
 <% provide :title, t(".home_title") %> 
 <h1><%= t ".heading" %></h1>
 <p>
   <%= t ".description_title" %>
   <%= link_to t("ruby_on_rails_tut"), root_path %> 
   <%= t ".description_detail" %>
 </p>
+
+<%= link_to "Sign up now!", signup_path, class: "btn btn-lg btn-primary" %>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: indenting 2 spaces for this line.

File: config/routes.rb [hunk 8]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
+  get 'users/new'
```
Comment (saobv-2223) [inline config/routes.rb]: using " instead of ' and move it to scope locale

Comment (EvonneNgXinYi) [reply config/routes.rb]: done

File: app/views/static_pages/home.html.erb (hunk 7 above)
Comment (EvonneNgXinYi) [reply app/views/static_pages/home.html.erb]: done

File: app/views/layouts/_footer.html.erb (hunk 6 above)
Comment (EvonneNgXinYi) [reply app/views/layouts/_footer.html.erb]: done

**Comment by EvonneNgXinYi** [comment]
ready

File: app/assets/stylesheets/variables.scss [hunk 9]
```diff
@@ -0,0 +1,5 @@
+$medium-light-gray: #eaeaea;
+$dark-gray: #222;
+$gray: #555;
+$light-gray: #777;
+white: #fff;
```
Comment (saobv-2223) [inline app/assets/stylesheets/variables.scss]:
```suggestion
$white: #fff;
```

**Review by saobv-2223 — COMMENTED** [review]
Have a small mistake about syntax, fix it and merge

**Pull request merged**